src/MassTransit/Internals/Extensions/AsyncEnumerableExtensions.cs

```diff
@@ -1,16 +1,17 @@
 namespace MassTransit.Internals
 {
     using System.Collections.Generic;
+    using System.Threading;
     using System.Threading.Tasks;
 
 
     public static class AsyncEnumerableExtensions
     {
-        public static async Task<IList<TElement>> ToListAsync<TElement>(this IAsyncEnumerable<TElement> elements)
+        public static async Task<IList<TElement>> ToListAsync<TElement>(this IAsyncEnumerable<TElement> elements, CancellationToken cancellationToken = default)
             where TElement : class
         {
             var elementsList = new List<TElement>();
-            await foreach (var element in elements.ConfigureAwait(false))
+            await foreach (var element in elements.WithCancellation(cancellationToken).ConfigureAwait(false))
```

Comment:
the overhead of this vs having two methods, I'd rather have two separate methods.